### ISSUE TITLE
[java] SuspiciousEqualsMethodName will detect the method with two input parameters

### ISSUE THREAD

**Issue by wuchiuwong** [pmd/pmd#3813]
**Affects PMD Version:** 
6.22.0

**Rule:**
SuspiciousEqualsMethodName

**Description:**
SuspiciousEqualsMethodName detect false-positive in the case: <code>boolean equals(Object foo, Object bar) {</code>
An equals method with two input parameters is unlikely to be an overriding equals method.

**Code Sample demonstrating the issue:**
```
public boolean equals(final Object a, final Object b) {
```

**Expected outcome:**
false-positive

**Running PMD through:**
CLI

**Comment by adangel** [comment]
I can confirm, that the rule will create a violation for your example.
However, the rule explicitly checks for exactly this case (equals with 2 parameters, at least one is of type Object) - although not explicitly documented, but it is in the example:

```java
   public boolean equals(Object o1, Object o2) {
     // oops, this probably was supposed to be equals(Object)
   }
```

What is correct now?

I think, this method is still suspicious equals - because it is called equals. If you program and use autocompletion, you might accidentally end up calling this method with two parameters which compares maybe against something different than "this" (what you maybe have expected). Since both parameters are Object, you don't even get a compiler warning if you call it with wrong/unexpected types.
